### PR TITLE
fix: Add support for borderRadii on RectButton

### PR DESCRIPTION
**Pull request by kacperkapusciak**
# Description

https://github.com/software-mansion/react-native-gesture-handler/assets/39658211/4ebc5761-c571-4c9a-a9c0-c2e0b7ac0018

This PR adds support for border radius style props like: `borderTopLeftRadius`, `borderTopRightRadius`, `borderBottomLeftRadius`, `borderBottomRightRadius` to RectButton.

It implements the border radii by wrapping a RawButton with two Views - the outer one curves the native RawButton and the inner acts like a mask cutting the corners of the button. The radius of the inner view had to be adjusted by subtracting the borderWidth from it - [see this tweet](https://twitter.com/lilykonings/status/1567317037126680576).

FIxes https://github.com/software-mansion/react-native-gesture-handler/issues/2594

### CODE REVIEW

**Review by m-bert — APPROVED** [review]
Logic looks fine, thanks for fixing that! ❤️  

I'll leave review of TS part for @tjzel

**Review by tjzel — APPROVED** [review]
TS seems approvable now 🚀

File: src/components/splitStyleProp.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    const innerBorderRadiiStyles = shrinkBorderRadiiByBorderWidth(
+      { ...styleGroups.borderRadiiStyles },
+      borderWidth
+    );
+
+    innerStyles = { ...innerStyles, ...innerBorderRadiiStyles };
+  }
+
+  return { outerStyles, innerStyles, restStyles };
+}
```
Comment (j-piasecki) [inline src/components/splitStyleProp.ts:169]: 1. Umm, do we care about `borderTopWidth`, `borderRightWidth`, ...?
2. It's better that now, so should we?

**Review by j-piasecki — CHANGES_REQUESTED** [review]

File: src/components/splitStyleProp.ts (hunk 1 above)
Comment (j-piasecki) [reply src/components/splitStyleProp.ts:169]: I'll assume that this comment being ignored by @kacperkapusciak means something along the lines of `and who is that? mr. grouchy, destroyer of good fun, slayer of children's smiles` and we'll figure it out in the due time.

**Review by j-piasecki — APPROVED** [review]
🔥

**Comment by versten1uk** [comment]
Still reproduced in `2.24.0`

**Pull request merged**